### ISSUE TITLE
occ_density_batch vs mask_camera

### ISSUE THREAD

**Issue by zzp0617** [cskkxjk/Vampire#3]
Hello, I found the following method to calculate occ_preds
```
a = occ_logits_batch[0] * occ_density_batch[0]
occ_preds = a.argmax(-1).cpu().detach().numpy()
```
I understand occ_logits_batch was trained by mask_camera, mask_camera consists of 0 and 1，But the trained occ_logits_batch consists of floating point numbers between 0 and 1，So I want to know how to compute the free of occ_pred.
I find that the calculation method of occ_result in validation_step is
```
occ_semantics = occ_semantics[mask_camera]
occ_logits_batch = occ_logits_batch[mask_camera]
# occ_density_batch = occ_density_batch[mask_camera]
# occ_free = (occ_density_batch == 0).float()
# occ_logits = torch.cat((occ_logits_batch, occ_free * 1e8), dim=1)
# occ_result = occ_logits.argmax(1)
occ_result = occ_logits_batch.argmax(1)
self.occ_val_iou(occ_result, occ_semantics)
```
It uses mask_camera to participate in the computation instead of occ_density_batch。
So I want to know how to use occ_density_batch to predict free, Can you give me some advice?

**Comment by cskkxjk** [comment]
I can't recall it precisely. However, it seems that the validation benchmark doesn't entail the evaluation of occ_free. Perhaps you can obtain occ_free by setting a value to compare with occ_density_batch.

**Comment by zzp0617** [comment]
Thanks for your reply, I use surroundocc methods to generate occ truth values on my own dataset, But surroundocc method is actually on the lidar coordinate system. But the code uses nuscenes ego coordinate system。
I tried to use my own generated occ truth values to train and found that the number of labels in number 17(free) was 0,（The number that was originally free is all in the number of the last target）.
but there is a number of 17(free) trained in the nuscenes dataset. Is it because of the coordinate system？ Can you give me some more advice?

**Comment by cskkxjk** [comment]
Hi~, can you describe your occ ground-truth generation method in details?

**Comment by zzp0617** [comment]
Occ ground-truth generation method is using surroundocc generation methods, Pointcloud multi-frame dense splicing ，The parameters are as follows:
'depth': 10   # 10
'min_density': 0.1
'n_threads': -1
'downsample': False
'voxel_size': 0.4
'max_nn': 20
'pc_range':  [-40, -40, -3.4, 40, 40, 3]
'occ_size':  [200, 200, 16]
'self_range': [3.0, 3.0, 3.0]

The pc_range of nuscenes is [-40, -40, -1, 40, 40, 5.4]， because nuscenes is in the ego coordinate system, my own occ ground-truth is in the lidar coordinate system, so the z value is -3.4m.
If possible, could you add a contact information， my wechat is: zzp_626
